063-prometheus-metrics-reporter.md

```diff
@@ -0,0 +1,208 @@
+# Prometheus Metrics Reporter
+
+I propose updating the way Kafka metrics are collected by implementing a metrics reporter that directly exposes metrics in the Prometheus format.
+
+PoC: https://github.com/mimaison/prometheus-metrics-reporter
+
+
+## Current situation
+
+Metrics are a critical aspect of monitoring Kafka. Today, this is the way metrics from Kafka are collected. First Kafka creates metrics objects in memory. For historical reasons Kafka uses two different libraries for metrics:
+- a home grown library, `org.apache.kafka.common.metrics`. We’ll refer to metrics created via this as _KafkaMetrics_. This is used on the client side and for common metrics on the server side too.
+- Yammer, `com.yammer.metrics.metrics-core`. This library is the predecessor of [Dropwizard](https://metrics.dropwizard.io/). We’ll refer to metrics created via this as _YammerMetrics_. This is only used on the broker side.
```

Comment:
```suggestion
- `org.apache.kafka.common.metrics` creates _Kafka metrics_ on the client side and common metrics on the server side.
- `com.yammer.metrics.metrics-core`) is the predecessor of [Dropwizard](https://metrics.dropwizard.io/) and creates _Yammer metrics_ on the broker side.
```

Reply:
I think its clearer just to say _Kafka metrics_ and _Yammer metrics_, as has been done elsewhere in the proposal.

Reply:
I did not apply your suggestion as it includes an extra parenthesis, but I committed a change that uses the same wording.